jdaviz/tests/test_subsets.py

```diff
@@ -951,3 +951,28 @@ def test_no_wcs_sky_regions(self, imviz_helper):
         imviz_helper.app.get_viewer('imviz-0').apply_roi(CircularROI(25, 25, 10))
         subsets = imviz_helper.app.get_subsets(include_sky_region=True)
         assert subsets['Subset 1'][0]['sky_region'] is None
+
+    def test_subset_renaming(self, specviz_helper, spectrum1d):
+        specviz_helper.load_data(spectrum1d, 'myfile')
+        spectrum_viewer_name = specviz_helper._default_spectrum_viewer_reference_name
+        viewer = specviz_helper.app.get_viewer(spectrum_viewer_name)
+
+        viewer.apply_roi(XRangeROI(6200, 6800))
+
+        get_data_1 = specviz_helper.get_data('myfile', spectral_subset='Subset 1')
+
+        subset_group = specviz_helper.app.data_collection.subset_groups
+        subset_group[0].label = 'diffname'
+        get_data_2 = specviz_helper.get_data('myfile', spectral_subset='diffname')
+
+        assert_quantity_allclose(get_data_1.flux, get_data_2.flux)
+        assert_quantity_allclose(get_data_1.spectral_axis, get_data_2.spectral_axis)
+
+        # when we officially implement subset renaming in jdaviz, we will want
+        # to be aware of this case and handle it appropriately
+        subset_group = specviz_helper.app.data_collection.subset_groups
+        subset_group[0].label = 'myfile'
+        get_data_3 = specviz_helper.get_data('myfile', spectral_subset='myfile')
+
+        assert_quantity_allclose(get_data_1.flux, get_data_3.flux)
+        assert_quantity_allclose(get_data_1.spectral_axis, get_data_3.spectral_axis)
```

Comment:
Can we also check that this is actually not the same as the data without the subset applied, just to be very sure?

Reply:
This is a good point, I should be checking the mask instead of flux or spectral axis to see if the subset is applied the same way 🤦‍♂️. Thank you for the reminder!

Reply:
@pllim Should be ready for a review now.